addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -125,65 +169,83 @@ export class DeletePlugin extends Plugin {
+    preserveCursor({ anchorNode, anchorOffset }) {
+        let cursorNode;
+        if (anchorOffset < nodeSize(anchorNode)) {
+            cursorNode = anchorNode.childNodes[anchorOffset];
+        }
+        return () => {
+            if (!cursorNode?.isConnected) {
+                // Fallback to original position
+                return { anchorNode, anchorOffset };
+            }
+            return {
+                anchorNode: cursorNode.parentElement,
+                anchorOffset: childNodeIndex(cursorNode),
+            };
+        };
+    }
+
+    splitTextNodes({ startContainer, startOffset, endContainer, endOffset }) {
+        // Splits text nodes only if necessary.
+        const split = (textNode, offset) => {
+            let didSplit = false;
+            if (offset === 0) {
+                offset = childNodeIndex(textNode);
+            } else if (offset === nodeSize(textNode)) {
+                offset = childNodeIndex(textNode) + 1;
+            } else {
+                textNode.splitText(offset);
+                didSplit = true;
+                offset = childNodeIndex(textNode) + 1;
+            }
+            return [textNode.parentElement, offset, didSplit];
+        };
+
         if (endContainer.nodeType === Node.TEXT_NODE) {
-            const newNode = endContainer.splitText(endOffset);
-            [endContainer, endOffset] = [newNode.parentElement, childNodeIndex(newNode)];
+            [endContainer, endOffset] = split(endContainer, endOffset);
         }
         if (startContainer.nodeType === Node.TEXT_NODE) {
-            const newNode = startContainer.splitText(startOffset);
-            [startContainer, startOffset] = [newNode.parentElement, childNodeIndex(newNode)];
-            if (startContainer === endContainer) {
+            let didSplit;
+            [startContainer, startOffset, didSplit] = split(startContainer, startOffset);
+            if (startContainer === endContainer && didSplit) {
                 endOffset += 1;
             }
         }
-        if (commonAncestorContainer.nodeType === Node.TEXT_NODE) {
-            commonAncestorContainer = commonAncestorContainer.parentElement;
-        }
 
         return {
             startElement: startContainer,
             startOffset,
             endElement: endContainer,
             endOffset,
-            commonAncestor: commonAncestorContainer,
         };
     }
 
     removeNodes({ startElement, startOffset, endElement, endOffset, commonAncestor }) {
         // Remove child nodes to the right, propagate until commonAncestor (non-inclusive)
         let node = startElement;
         let startRemoveIndex = startOffset;
+        const nodesToRemove = [];
         while (node !== commonAncestor) {
-            // @phoenix @todo: handle unremovable nodes
-            [...node.childNodes].slice(startRemoveIndex).forEach((child) => child.remove());
+            for (const child of [...node.childNodes].slice(startRemoveIndex)) {
```

Comment:
i know it's not a big deal, but i don't like creating all the time arrays, then slicing them. it feels like making the browser work too much for no good reason. also, unless i'm mistaken, the basic for loop is not much more complicated:
```js
for (let i = 0; i < startRemoveIndex; i++) {
  nodesToRemove.push(node.childNodes[i]);
}
```
(just need to make sure i am correct with the index)

Reply:
done.